--- mcrouter/mcrouter/m4/ax_boost_context.m4~	2015-04-01 04:20:13.000000000 +0300
+++ mcrouter/mcrouter/m4/ax_boost_context.m4	2015-04-06 13:47:27.995041619 +0300
@@ -70,7 +70,7 @@
 
       AC_COMPILE_IFELSE([AC_LANG_PROGRAM(
         [[@%:@include <boost/context/all.hpp>]],
-        [[boost::context::fcontext_t* fc = boost::context::make_fcontext(0, 0, 0);]])],
+        [[boost::context::fcontext_t fc = boost::context::make_fcontext(0, 0, 0);]])],
         ax_cv_boost_context=yes, ax_cv_boost_context=no)
         CXXFLAGS=$CXXFLAGS_SAVE
       AC_LANG_POP([C++])
